docs/dev/guidelines/database.rst

```diff
@@ -142,6 +142,70 @@ Entity relationships often depend on the existence of another entity — for exa
 Not used in Artemis yet.
 
 
+4. Dynamic Fetching
+===================
+In Artemis, we use dynamic fetching to load the relationships of an entity on demand. As we do not want to load all relationships of an entity every time we fetch it from the database, we use as described above ``FetchType.LAZY``.
+In order to load the relationships of an entity on demand, we then use one of 3 methods:
+
+1. **EntityGraph**: The ``@EntityGraph`` annotation is the simples way to specify a graph of relationships to fetch. It should be used when a query is auto-constructed by Spring Data JPA and does not have a custom ``@Query`` annotation. Example:
+
+ .. code-block:: java
+
+    // CourseRepository.java
+    @EntityGraph(type = LOAD, attributePaths = { "exercises", "exercises.categories", "exercises.teamAssignmentConfig" })
+    Course findWithEagerExercisesById(long courseId);
+
+2. **JOIN FETCH**: The ``JOIN FETCH`` keyword is used in a custom query to specify a graph of relationships to fetch. It should be used when a query is custom and has a custom ``@Query`` annotation. Example:
+
+ .. code-block:: java
+
+    // ProgrammingExerciseRepository.java
+    @Query("""
+            SELECT pe
+            FROM ProgrammingExercise pe
+                LEFT JOIN FETCH pe.exerciseGroup eg
+                LEFT JOIN FETCH eg.exam e
+            WHERE e.endDate > :dateTime
+            """)
+    List<ProgrammingExercise> findAllWithEagerExamByExamEndDateAfterDate(@Param("dateTime") ZonedDateTime dateTime);
```

Comment:
```suggestion
    Set<ProgrammingExercise> findAllWithEagerExamByExamEndDateAfterDate(@Param("dateTime") ZonedDateTime dateTime);
```

Reply:
This was taken from real code, where it is in fact `List` 🤷🏻